app/src/main/java/com/nervousfish/nervousfish/activities/NFCActivity.java

```diff
@@ -1,22 +1,147 @@
+import android.nfc.NfcAdapter;
+import android.nfc.NfcAdapter.CreateNdefMessageCallback;
+import android.nfc.NfcEvent;
 import android.os.Bundle;
+import android.os.Parcelable;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.Toast;
 
+import com.bumptech.glide.Glide;
+import com.nervousfish.nervousfish.ConstantKeywords;
 import com.nervousfish.nervousfish.R;
+import com.nervousfish.nervousfish.data_objects.Contact;
+import com.nervousfish.nervousfish.data_objects.Profile;
+import com.nervousfish.nervousfish.data_objects.SimpleKey;
+import com.nervousfish.nervousfish.service_locator.IServiceLocator;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+import static android.nfc.NdefRecord.createExternal;
 
 /**
- * An {@link Activity} that used for pairing over NFC
+ * An {@link Activity} that beams NDEF Messages to Other Devices.
  */
-public class NFCActivity extends Activity {
+public class NFCActivity extends Activity implements CreateNdefMessageCallback {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger("NFCActivity");
+    private TextView textView;
+    private IServiceLocator serviceLocator;
+    private byte[] bytes;
+
     /**
-     * Creates the new activity, should only be called by Android
-     *
-     * @param savedInstanceState The saved state of the instance.
+     * {@inheritDoc}
      */
     @Override
-    protected void onCreate(final Bundle savedInstanceState) {
+    public void onCreate(final Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_nfc);
+
+        final Intent intent = getIntent();
+        this.serviceLocator = (IServiceLocator) intent.getSerializableExtra(ConstantKeywords.SERVICE_LOCATOR);
+
+        textView = (TextView) findViewById(R.id.nfc_instruction);
+        // Check for available NFC Adapter
+        final NfcAdapter nfcAdapter = NfcAdapter.getDefaultAdapter(this);
+        if (nfcAdapter == null) {
+            Toast.makeText(this, "NFC is not available", Toast.LENGTH_LONG).show();
+            finish();
+            return;
+        }
+        LOGGER.info("Start creating an NDEF message to beam");
+        Glide.with(this).load(R.drawable.s_contact_animado).into((ImageView) findViewById(R.id.nfc_gif));
+        try {
+            final Profile myProfile = this.serviceLocator.getDatabase().getProfiles().get(0);
+            LOGGER.info("Sending my profile with name: " + myProfile.getName() + ", public key: "
+                    + myProfile.getPublicKey().toString());
+            final Contact contact = new Contact(myProfile.getName(), new SimpleKey("simplekey", "73890ien"));
+            bytes = this.serviceLocator.getNFCHandler().objectToBytes(contact);
+        } catch (IOException e) {
+            LOGGER.error("Could not serialize my contact to other device " + e.getMessage());
```

Comment:
Now you just throw the full stacktrace of e away, you can better use
`LOGGER.error("Could not serialize my contact to other device", e);`